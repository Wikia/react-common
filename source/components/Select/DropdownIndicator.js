import React from 'react';
import { components } from 'react-select';

export const SearchDropdownIndicator = props => (
    <components.DropdownIndicator {...props}>
        <svg
            className="search-dropdown-indicator"
            xmlns="http://www.w3.org/2000/svg"
            viewBox="0 0 18 18"
        >
            <g fillRule="evenodd">
                <path d="M16.984 16.025l-4.03-4.043a.713.713 0 0 0-1.011 0 .72.72 0 0 0 0 1.015l4.03 4.043c.279.28.732.28 1.011 0a.72.72 0 0 0 0-1.015z" />
                <path d="M2.178 7.924c0-3.17 2.56-5.74 5.72-5.74 3.16 0 5.72 2.57 5.72 5.74 0 3.17-2.56 5.739-5.72 5.739-3.16 0-5.72-2.57-5.72-5.74zm-1.43 0c0 3.962 3.2 7.174 7.15 7.174s7.15-3.212 7.15-7.174S11.848.75 7.898.75.748 3.962.748 7.924z" />
            </g>
        </svg>
    </components.DropdownIndicator>
);

export const DefaultDropdownIndicator = (props) => {
    const className = [
        'default-dropdown-indicator',
        props.isFocused ? 'is-focused' : undefined,
    ]
        .filter(Boolean)
        .join(' ');

    return (
        <components.DropdownIndicator {...props}>
            <svg
                className={className}
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
            >
                <path d="M7 8h10l-5 6z" />
            </svg>
        </components.DropdownIndicator>
    );
};
