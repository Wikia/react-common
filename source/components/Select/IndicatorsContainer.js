import React from 'react';
import { components } from 'react-select';

const IndicatorsContainer = (props) => {
    const { isLoading } = props.selectProps;
    // eslint-disable-next-line no-unused-vars
    const [clearIndicator, spinner, indicatorSeparator, dropdownIndicator] = props.children;

    return (
        <components.IndicatorsContainer {...props}>
            {spinner}
            {isLoading ? null : dropdownIndicator}
            {indicatorSeparator}
        </components.IndicatorsContainer>
    );
};

export default IndicatorsContainer;
