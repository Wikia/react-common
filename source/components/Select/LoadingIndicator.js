import React from 'react';

// eslint-disable-next-line no-restricted-imports
import Spinner from '../Spinner';

const LoadingIndicator = () => (
    <Spinner size={24} />
);

export default LoadingIndicator;
