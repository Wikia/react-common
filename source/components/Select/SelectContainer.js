import React from 'react';
import { components } from 'react-select';

const SelectContainer = props => (
    <div className={`${props.selectProps.classNamePrefix}__wrapper`}>
        <components.SelectContainer {...props} />
    </div>
);

export default SelectContainer;
