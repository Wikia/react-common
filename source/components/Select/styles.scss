@import "~design-system/dist/scss/wds-variables/index.scss";

.fandom-select__wrapper {
	background-color: $wds-color-white;
	border: 1px solid $wds-color-light-gray;
	border-radius: 4px;
	padding: 5px 10px;

	// it's odd to have the separate selector here but it's needed for specificity to avoid !important
	.fandom-select {
		&__control {
			background-color: inherit;
			border: none;
			border-radius: 0;
			box-shadow: none;
			min-height: 0;

			&--is-disabled {
				background-color: inherit;
			}
		}

		&__value-container {
			min-height: 27px;
			padding: 0;
		}

		&__dropdown-indicator {
			padding: 0;

			> .default-dropdown-indicator {
				fill: currentColor;
				height: 24px;
				width: 24px;

				&.is-focused {
					transform: rotate(180deg);
				}
			}

			> .search-dropdown-indicator {
				fill: currentColor;
				height: 16px;
				width: 16px;
			}
		}

		&__menu {
			border-top: 1px solid $wds-color-black;
			border-top-left-radius: 0;
			border-top-right-radius: 0;
			margin-top: 0;
			z-index: $z-max;
		}
	}
}
