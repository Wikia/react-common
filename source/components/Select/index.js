import React from 'react';
import PropTypes from 'prop-types';
import ReactSelect from 'react-select';

import './styles.scss';
import IndicatorsContainer from './IndicatorsContainer';
import SelectContainer from './SelectContainer';
import { DefaultDropdownIndicator, SearchDropdownIndicator } from './DropdownIndicator';
import LoadingIndicator from './LoadingIndicator';

function callWithValues(func, values, isMulti) {
    if (!func) {
        return;
    }

    if (isMulti) {
        func(values);
        return;
    }

    if (values.length === 0) {
        func(null, null);
        return;
    }

    const { value, label } = values[0];
    func(value, label);
}

class Select extends React.Component {
    static createOption(value, label) {
        return { value, label };
    }

    constructor(props) {
        super(props);
        this.selectRef = React.createRef();
        this.onRequestNoOptionsMessage = this.onRequestNoOptionsMessage.bind(this);
        this.onBlur = this.onBlur.bind(this);
        this.onChange = this.onChange.bind(this);
        this.onFocus = this.onFocus.bind(this);
        this.onTextInputChange = this.onTextInputChange.bind(this);
    }

    onBlur() {
        if (!this.props.onBlur || !this.selectRef.current) {
            return;
        }

        callWithValues(this.props.onBlur, this.selectRef.current.select.getCommonProps().getValue(), this.props.multi);
    }

    onChange(values) {
        const valuesAsArray = this.props.multi ? values : [values];
        callWithValues(this.props.onChange, valuesAsArray, this.props.multi);
    }

    onFocus() {
        if (!this.props.onFocus) {
            return;
        }

        this.props.onFocus();
    }

    onRequestNoOptionsMessage({ inputValue }) {
        const { onNoOptions } = this.props;
        if (typeof onNoOptions === 'function') {
            return onNoOptions(inputValue);
        }

        return onNoOptions;
    }

    onTextInputChange(input) {
        if (this.props.onTextInputChange) {
            this.props.onTextInputChange(input);
        }
    }

    getRootClassName() {
        return ['fandom-select', this.props.className]
            .filter(Boolean)
            .join(' ');
    }

    getValueFromProps() {
        const { value, options } = this.props;

        if (!value) {
            return undefined;
        }

        let valuesWithLabels = value;
        if (!(value instanceof Array)) {
            valuesWithLabels = [value];
        }

        valuesWithLabels = valuesWithLabels
            .map(v => options.find(o => o.value === v))
            .filter(Boolean);

        if (valuesWithLabels.length === 0) {
            return undefined;
        }

        if (this.props.multi) {
            return valuesWithLabels;
        }

        return valuesWithLabels[0];
    }

    render() {
        return (
            <ReactSelect
                ref={this.selectRef}
                autoFocus={this.props.autoFocus}
                blurInputOnSelect
                className={this.getRootClassName()}
                classNamePrefix="fandom-select"
                controlShouldRenderValue={this.props.multi ? this.props.multiValueRender : true}
                isDisabled={this.props.disabled || this.props.loading}
                isLoading={this.props.loading}
                isMulti={this.props.multi}
                isSearchable={this.props.searchable}
                noOptionsMessage={this.onRequestNoOptionsMessage}
                placeholder={this.props.placeholder}
                onBlur={this.onBlur}
                onChange={this.onChange}
                onFocus={this.onFocus}
                onInputChange={this.onTextInputChange}
                options={this.props.options}
                value={this.getValueFromProps()}
                components={{
                    IndicatorSeparator: null,
                    DropdownIndicator: this.props.searchable ? SearchDropdownIndicator : DefaultDropdownIndicator,
                    LoadingIndicator,
                    IndicatorsContainer,
                    SelectContainer,
                }}
            />
        );
    }
}

Select.propTypes = {
    /**
     * Focus the control when it is mounted
     */
    autoFocus: PropTypes.bool,
    /**
     * Additional class for root element
     */
    className: PropTypes.string,
    /**
     * whether or not the input is disabled
     */
    disabled: PropTypes.bool,
    /**
     * whether or not to show loading indicator
     */
    loading: PropTypes.bool,
    /**
     * whether or not multiple values are allowed
     */
    multi: PropTypes.bool,
    /**
     * whether or not the component should render values when `multi=true`
     */
    multiValueRender: PropTypes.bool,
    /**
     * called when the input is blurred `onBlur(val, label)`
     */
    onBlur: PropTypes.func,
    /**
     * called when the input is changed
     */
    onChange: PropTypes.func,
    /**
     * called when input is focused `onFocus()`
     */
    onFocus: PropTypes.func,
    /**
     * message when no options are present. If a `func`, the only argument is the input to the text input
     */
    onNoOptions: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.func,
    ]),
    /**
     * fired when the search input is changed. `onTextInputChange(currentInput)`
     */
    onTextInputChange: PropTypes.func,
    /**
     * options to display. Use `createOption` exported from this module to create options
     */
    options: PropTypes.arrayOf(PropTypes.shape({
        label: PropTypes.string.isRequired,
        value: PropTypes.any.isRequired,
    })),
    /**
     * Placeholder text used when no value is selected
     */
    placeholder: PropTypes.string,
    /**
     * whether or not to allow text searching
     */
    searchable: PropTypes.bool,
    /**
     * when using as controlled input, the currently selected values
     */
    value: PropTypes.oneOfType([
        PropTypes.arrayOf(PropTypes.any),
        PropTypes.any,
    ]),
};

Select.defaultProps = {
    autoFocus: false,
    className: '',
    disabled: false,
    loading: false,
    multi: false,
    multiValueRender: true,
    onBlur: undefined,
    onChange: undefined,
    onFocus: undefined,
    onNoOptions: 'No Options',
    onTextInputChange: undefined,
    options: [],
    placeholder: 'Select...',
    searchable: true,
    value: undefined,
};

export default Select;
