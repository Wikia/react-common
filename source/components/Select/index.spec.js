import React from 'react';
import renderer from 'react-test-renderer';
import { components } from 'react-select';
import {
    mount,
} from 'enzyme';
import sinon from 'sinon';

import { DefaultDropdownIndicator } from './DropdownIndicator';

import Select from './index';

test('Select renders correctly', () => {
    let component = renderer.create(
        <Select
            options={[
                Select.createOption(1, 'label1'),
                Select.createOption(2, 'label2'),
                Select.createOption(3, 'label3'),
            ]}
        />
    );
    expect(component.toJSON()).toMatchSnapshot();

    component = renderer.create(
        <Select
            value={1}
            options={[
                Select.createOption(1, 'label1'),
                Select.createOption(2, 'label2'),
                Select.createOption(3, 'label3'),
            ]}
        />
    );
    expect(component.toJSON()).toMatchSnapshot();

    component = renderer.create(
        <Select
            value={[1, 3]}
            multi
            options={[
                Select.createOption(1, 'label1'),
                Select.createOption(2, 'label2'),
                Select.createOption(3, 'label3'),
            ]}
        />
    );
    expect(component.toJSON()).toMatchSnapshot();

    component = renderer.create(
        <Select
            value={4}
            options={[
                Select.createOption(1, 'label1'),
                Select.createOption(2, 'label2'),
                Select.createOption(3, 'label3'),
            ]}
        />
    );
    expect(component.toJSON()).toMatchSnapshot();

    component = renderer.create(<Select options={[]} />);
    expect(component.toJSON()).toMatchSnapshot();

    component = renderer.create(<Select searchable={false} />);
    expect(component.toJSON()).toMatchSnapshot();

    component = renderer.create(<Select loading />);
    expect(component.toJSON()).toMatchSnapshot();

    component = renderer.create(
        <DefaultDropdownIndicator
            cx={() => {}}
            getStyles={() => {}}
            isFocused
        />
    );
    expect(component.toJSON()).toMatchSnapshot();
});

test('Select onBlur functionality', () => {
    const onBlurSpy = sinon.spy();
    let wrapper = mount(
        <Select
            onBlur={onBlurSpy}
            options={[
                Select.createOption(1, 'label1'),
                Select.createOption(2, 'label2'),
                Select.createOption(3, 'label3'),
            ]}
        />
    );

    let stateManager = wrapper.find('StateManager').at(0);
    let input = wrapper.find('.fandom-select__input input').at(0);

    input.simulate('blur');
    expect(onBlurSpy.withArgs(null, null).calledOnce).toBe(true);

    stateManager.setState({ menuIsOpen: true });

    let option = wrapper.find(components.Option).at(1).get(0);
    option.props.innerProps.onClick();
    input.simulate('blur');
    expect(onBlurSpy.withArgs(2, 'label2').calledOnce).toBe(true);

    wrapper = mount(
        <Select
            options={[
                Select.createOption(1, 'label1'),
                Select.createOption(2, 'label2'),
                Select.createOption(3, 'label3'),
            ]}
        />
    );

    stateManager = wrapper.find('StateManager').at(0);
    input = wrapper.find('.fandom-select__input input').at(0);
    stateManager.setState({ menuIsOpen: true });
    option = wrapper.find(components.Option).at(1).get(0);
    option.props.innerProps.onClick();
    input.simulate('blur');
});

test('Multi-Select tests', () => {
    const onBlurSpy = sinon.spy();
    const wrapper = mount(
        <Select
            multi
            onBlur={onBlurSpy}
            options={[
                Select.createOption(1, 'label1'),
                Select.createOption(2, 'label2'),
                Select.createOption(3, 'label3'),
            ]}
        />
    );

    const stateManager = wrapper.find('StateManager').at(0);
    const input = wrapper.find('.fandom-select__input input').at(0);

    stateManager.setState({ menuIsOpen: true });
    const option1 = wrapper.find(components.Option).at(0).get(0);
    option1.props.innerProps.onClick();

    stateManager.setState({ menuIsOpen: false });
    input.simulate('blur');

    expect(onBlurSpy.withArgs([{ value: 1, label: 'label1' }]).calledOnce).toBe(true);

    stateManager.setState({ menuIsOpen: true });
    const option2 = wrapper.find(components.Option).at(0).get(0);
    option2.props.innerProps.onClick();
    stateManager.setState({ menuIsOpen: false });
    input.simulate('blur');
    expect(onBlurSpy.withArgs([{ value: 1, label: 'label1' }, { value: 2, label: 'label2' }]).calledOnce).toBe(true);
});

test('onNoOptions tests', () => {
    let wrapper = mount(
        <Select
            onNoOptions="‚òπÔ∏è"
            options={[]}
        />
    );

    let stateManager = wrapper.find('StateManager').at(0);
    stateManager.setState({ menuIsOpen: true });
    const optionRenderer = wrapper.find('.fandom-select__menu-notice--no-options');
    expect(optionRenderer.text()).toEqual('‚òπÔ∏è');

    const onNoOptionsSpy = sinon.spy();
    wrapper = mount(
        <Select
            onNoOptions={onNoOptionsSpy}
            options={[]}
        />
    );

    stateManager = wrapper.find('StateManager').at(0);
    stateManager.setState({ menuIsOpen: true });
    expect(onNoOptionsSpy.calledOnce).toBe(true);
});

test('onFocus tests', () => {
    const onFocusSpy = sinon.spy();
    let wrapper = mount(<Select onFocus={onFocusSpy} />);
    let input = wrapper.find('.fandom-select__input input').at(0);
    input.simulate('focus');

    expect(onFocusSpy.calledOnce).toBe(true);

    wrapper = mount(<Select />);
    input = wrapper.find('.fandom-select__input input').at(0);
    input.simulate('focus');
});

test('onTextInputChange tests', () => {
    const onTextInputChangeSpy = sinon.spy();
    const wrapper = mount(<Select onTextInputChange={onTextInputChangeSpy} />);
    const input = wrapper.find(components.Input).at(0).get(0);

    input.props.onChange({ currentTarget: { value: 'üòÄ' } });
    expect(onTextInputChangeSpy.withArgs('üòÄ').calledOnce).toBe(true);
});

test('custom className is at root level', () => {
    const wrapper = mount(<Select className="custom-select" />);
    expect(wrapper.find('.fandom-select__wrapper.custom-select')).toHaveLength(1);
});
