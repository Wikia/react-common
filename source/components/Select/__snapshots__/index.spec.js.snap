// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders correctly 1`] = `
<div
  className="fandom-select__wrapper"
>
  <div
    className="fandom-select css-2b097c-container"
    onKeyDown={[Function]}
  >
    <div
      className="fandom-select__control css-yk16xz-control"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
    >
      <div
        className="fandom-select__value-container css-1hwfws3"
      >
        <div
          className="fandom-select__placeholder css-1wa3eu0-placeholder"
        >
          Select...
        </div>
        <div
          className="css-b8ldur-Input"
        >
          <div
            className="fandom-select__input"
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <input
              aria-autocomplete="list"
              autoCapitalize="none"
              autoComplete="off"
              autoCorrect="off"
              disabled={false}
              id="fandom-select-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              spellCheck="false"
              style={
                Object {
                  "background": 0,
                  "border": 0,
                  "boxSizing": "content-box",
                  "color": "inherit",
                  "fontSize": "inherit",
                  "label": "input",
                  "opacity": 1,
                  "outline": 0,
                  "padding": 0,
                  "width": "1px",
                }
              }
              tabIndex="0"
              type="text"
              value=""
            />
            <div
              style={
                Object {
                  "height": 0,
                  "left": 0,
                  "overflow": "scroll",
                  "position": "absolute",
                  "top": 0,
                  "visibility": "hidden",
                  "whiteSpace": "pre",
                }
              }
            >
              
            </div>
          </div>
        </div>
      </div>
      <div
        className="fandom-select__indicators css-1hb7zxy-IndicatorsContainer"
      >
        <div
          aria-hidden="true"
          className="fandom-select__indicator fandom-select__dropdown-indicator css-tlfecz-indicatorContainer"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className="search-dropdown-indicator"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Select renders correctly 2`] = `
<div
  className="fandom-select__wrapper"
>
  <div
    className="fandom-select css-2b097c-container"
    onKeyDown={[Function]}
  >
    <div
      className="fandom-select__control css-yk16xz-control"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
    >
      <div
        className="fandom-select__value-container fandom-select__value-container--has-value css-1hwfws3"
      >
        <div
          className="fandom-select__single-value css-1uccc91-singleValue"
        >
          label1
        </div>
        <div
          className="css-b8ldur-Input"
        >
          <div
            className="fandom-select__input"
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <input
              aria-autocomplete="list"
              autoCapitalize="none"
              autoComplete="off"
              autoCorrect="off"
              disabled={false}
              id="fandom-select-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              spellCheck="false"
              style={
                Object {
                  "background": 0,
                  "border": 0,
                  "boxSizing": "content-box",
                  "color": "inherit",
                  "fontSize": "inherit",
                  "label": "input",
                  "opacity": 1,
                  "outline": 0,
                  "padding": 0,
                  "width": "1px",
                }
              }
              tabIndex="0"
              type="text"
              value=""
            />
            <div
              style={
                Object {
                  "height": 0,
                  "left": 0,
                  "overflow": "scroll",
                  "position": "absolute",
                  "top": 0,
                  "visibility": "hidden",
                  "whiteSpace": "pre",
                }
              }
            >
              
            </div>
          </div>
        </div>
      </div>
      <div
        className="fandom-select__indicators css-1hb7zxy-IndicatorsContainer"
      >
        <div
          aria-hidden="true"
          className="fandom-select__indicator fandom-select__dropdown-indicator css-tlfecz-indicatorContainer"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className="search-dropdown-indicator"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Select renders correctly 3`] = `
<div
  className="fandom-select__wrapper"
>
  <div
    className="fandom-select css-2b097c-container"
    onKeyDown={[Function]}
  >
    <div
      className="fandom-select__control css-yk16xz-control"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
    >
      <div
        className="fandom-select__value-container fandom-select__value-container--is-multi fandom-select__value-container--has-value css-1hwfws3"
      >
        <div
          className="css-1rhbuit-multiValue fandom-select__multi-value"
        >
          <div
            className="css-12jo7m5 fandom-select__multi-value__label"
          >
            label1
          </div>
          <div
            className="css-xb97g8 fandom-select__multi-value__remove"
            onClick={[Function]}
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <svg
              aria-hidden="true"
              className="css-6q0nyr-Svg"
              focusable="false"
              height={14}
              viewBox="0 0 20 20"
              width={14}
            >
              <path
                d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
              />
            </svg>
          </div>
        </div>
        <div
          className="css-1rhbuit-multiValue fandom-select__multi-value"
        >
          <div
            className="css-12jo7m5 fandom-select__multi-value__label"
          >
            label3
          </div>
          <div
            className="css-xb97g8 fandom-select__multi-value__remove"
            onClick={[Function]}
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <svg
              aria-hidden="true"
              className="css-6q0nyr-Svg"
              focusable="false"
              height={14}
              viewBox="0 0 20 20"
              width={14}
            >
              <path
                d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
              />
            </svg>
          </div>
        </div>
        <div
          className="css-b8ldur-Input"
        >
          <div
            className="fandom-select__input"
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <input
              aria-autocomplete="list"
              autoCapitalize="none"
              autoComplete="off"
              autoCorrect="off"
              disabled={false}
              id="fandom-select-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              spellCheck="false"
              style={
                Object {
                  "background": 0,
                  "border": 0,
                  "boxSizing": "content-box",
                  "color": "inherit",
                  "fontSize": "inherit",
                  "label": "input",
                  "opacity": 1,
                  "outline": 0,
                  "padding": 0,
                  "width": "1px",
                }
              }
              tabIndex="0"
              type="text"
              value=""
            />
            <div
              style={
                Object {
                  "height": 0,
                  "left": 0,
                  "overflow": "scroll",
                  "position": "absolute",
                  "top": 0,
                  "visibility": "hidden",
                  "whiteSpace": "pre",
                }
              }
            >
              
            </div>
          </div>
        </div>
      </div>
      <div
        className="fandom-select__indicators css-1hb7zxy-IndicatorsContainer"
      >
        <div
          aria-hidden="true"
          className="fandom-select__indicator fandom-select__dropdown-indicator css-tlfecz-indicatorContainer"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className="search-dropdown-indicator"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Select renders correctly 4`] = `
<div
  className="fandom-select__wrapper"
>
  <div
    className="fandom-select css-2b097c-container"
    onKeyDown={[Function]}
  >
    <div
      className="fandom-select__control css-yk16xz-control"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
    >
      <div
        className="fandom-select__value-container css-1hwfws3"
      >
        <div
          className="fandom-select__placeholder css-1wa3eu0-placeholder"
        >
          Select...
        </div>
        <div
          className="css-b8ldur-Input"
        >
          <div
            className="fandom-select__input"
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <input
              aria-autocomplete="list"
              autoCapitalize="none"
              autoComplete="off"
              autoCorrect="off"
              disabled={false}
              id="fandom-select-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              spellCheck="false"
              style={
                Object {
                  "background": 0,
                  "border": 0,
                  "boxSizing": "content-box",
                  "color": "inherit",
                  "fontSize": "inherit",
                  "label": "input",
                  "opacity": 1,
                  "outline": 0,
                  "padding": 0,
                  "width": "1px",
                }
              }
              tabIndex="0"
              type="text"
              value=""
            />
            <div
              style={
                Object {
                  "height": 0,
                  "left": 0,
                  "overflow": "scroll",
                  "position": "absolute",
                  "top": 0,
                  "visibility": "hidden",
                  "whiteSpace": "pre",
                }
              }
            >
              
            </div>
          </div>
        </div>
      </div>
      <div
        className="fandom-select__indicators css-1hb7zxy-IndicatorsContainer"
      >
        <div
          aria-hidden="true"
          className="fandom-select__indicator fandom-select__dropdown-indicator css-tlfecz-indicatorContainer"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className="search-dropdown-indicator"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Select renders correctly 5`] = `
<div
  className="fandom-select__wrapper"
>
  <div
    className="fandom-select css-2b097c-container"
    onKeyDown={[Function]}
  >
    <div
      className="fandom-select__control css-yk16xz-control"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
    >
      <div
        className="fandom-select__value-container css-1hwfws3"
      >
        <div
          className="fandom-select__placeholder css-1wa3eu0-placeholder"
        >
          Select...
        </div>
        <div
          className="css-b8ldur-Input"
        >
          <div
            className="fandom-select__input"
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <input
              aria-autocomplete="list"
              autoCapitalize="none"
              autoComplete="off"
              autoCorrect="off"
              disabled={false}
              id="fandom-select-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              spellCheck="false"
              style={
                Object {
                  "background": 0,
                  "border": 0,
                  "boxSizing": "content-box",
                  "color": "inherit",
                  "fontSize": "inherit",
                  "label": "input",
                  "opacity": 1,
                  "outline": 0,
                  "padding": 0,
                  "width": "1px",
                }
              }
              tabIndex="0"
              type="text"
              value=""
            />
            <div
              style={
                Object {
                  "height": 0,
                  "left": 0,
                  "overflow": "scroll",
                  "position": "absolute",
                  "top": 0,
                  "visibility": "hidden",
                  "whiteSpace": "pre",
                }
              }
            >
              
            </div>
          </div>
        </div>
      </div>
      <div
        className="fandom-select__indicators css-1hb7zxy-IndicatorsContainer"
      >
        <div
          aria-hidden="true"
          className="fandom-select__indicator fandom-select__dropdown-indicator css-tlfecz-indicatorContainer"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className="search-dropdown-indicator"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Select renders correctly 6`] = `
<div
  className="fandom-select__wrapper"
>
  <div
    className="fandom-select css-2b097c-container"
    onKeyDown={[Function]}
  >
    <div
      className="fandom-select__control css-yk16xz-control"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
    >
      <div
        className="fandom-select__value-container css-1hwfws3"
      >
        <div
          className="fandom-select__placeholder css-1wa3eu0-placeholder"
        >
          Select...
        </div>
        <input
          className="css-62g3xt-dummyInput"
          disabled={false}
          id="fandom-select-input"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          readOnly={true}
          tabIndex="0"
          value=""
        />
      </div>
      <div
        className="fandom-select__indicators css-1hb7zxy-IndicatorsContainer"
      >
        <div
          aria-hidden="true"
          className="fandom-select__indicator fandom-select__dropdown-indicator css-tlfecz-indicatorContainer"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className="default-dropdown-indicator"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Select renders correctly 7`] = `
<div
  className="fandom-select__wrapper"
>
  <div
    className="fandom-select fandom-select--is-disabled css-14jk2my-container"
    onKeyDown={[Function]}
  >
    <div
      className="fandom-select__control fandom-select__control--is-disabled css-1fhf3k1-control"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
    >
      <div
        className="fandom-select__value-container css-1hwfws3"
      >
        <div
          className="fandom-select__placeholder css-1wa3eu0-placeholder"
        >
          Select...
        </div>
        <div
          className="css-b8ldur-Input"
        >
          <div
            className="fandom-select__input"
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <input
              aria-autocomplete="list"
              autoCapitalize="none"
              autoComplete="off"
              autoCorrect="off"
              disabled={true}
              id="fandom-select-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              spellCheck="false"
              style={
                Object {
                  "background": 0,
                  "border": 0,
                  "boxSizing": "content-box",
                  "color": "inherit",
                  "fontSize": "inherit",
                  "label": "input",
                  "opacity": 1,
                  "outline": 0,
                  "padding": 0,
                  "width": "1px",
                }
              }
              tabIndex="0"
              type="text"
              value=""
            />
            <div
              style={
                Object {
                  "height": 0,
                  "left": 0,
                  "overflow": "scroll",
                  "position": "absolute",
                  "top": 0,
                  "visibility": "hidden",
                  "whiteSpace": "pre",
                }
              }
            >
              
            </div>
          </div>
        </div>
      </div>
      <div
        className="fandom-select__indicators css-1hb7zxy-IndicatorsContainer"
      >
        <div
          className="fandom-spinner"
          style={
            Object {
              "height": 24,
              "width": 24,
            }
          }
        >
          <svg
            height={24}
            viewBox="0 0 24 24"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              transform="translate(12, 12)"
            >
              <circle
                fill="none"
                r={11}
                strokeDasharray={69.11503837897544}
                strokeDashoffset={69.11503837897544}
                strokeLinecap="round"
                strokeWidth={2}
              />
            </g>
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Select renders correctly 8`] = `
<div
  className="css-hbt8bd-DropdownIndicator"
>
  <div
    className="default-dropdown-indicator is-focused"
  />
</div>
`;

exports[`Select renders with grouped options 1`] = `
<div
  className="fandom-select__wrapper"
>
  <div
    className="fandom-select css-2b097c-container"
    onKeyDown={[Function]}
  >
    <div
      className="fandom-select__control css-yk16xz-control"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
    >
      <div
        className="fandom-select__value-container fandom-select__value-container--has-value css-1hwfws3"
      >
        <div
          className="fandom-select__single-value css-1uccc91-singleValue"
        >
          label4
        </div>
        <div
          className="css-b8ldur-Input"
        >
          <div
            className="fandom-select__input"
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <input
              aria-autocomplete="list"
              autoCapitalize="none"
              autoComplete="off"
              autoCorrect="off"
              disabled={false}
              id="fandom-select-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              spellCheck="false"
              style={
                Object {
                  "background": 0,
                  "border": 0,
                  "boxSizing": "content-box",
                  "color": "inherit",
                  "fontSize": "inherit",
                  "label": "input",
                  "opacity": 1,
                  "outline": 0,
                  "padding": 0,
                  "width": "1px",
                }
              }
              tabIndex="0"
              type="text"
              value=""
            />
            <div
              style={
                Object {
                  "height": 0,
                  "left": 0,
                  "overflow": "scroll",
                  "position": "absolute",
                  "top": 0,
                  "visibility": "hidden",
                  "whiteSpace": "pre",
                }
              }
            >
              
            </div>
          </div>
        </div>
      </div>
      <div
        className="fandom-select__indicators css-1hb7zxy-IndicatorsContainer"
      >
        <div
          aria-hidden="true"
          className="fandom-select__indicator fandom-select__dropdown-indicator css-tlfecz-indicatorContainer"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className="search-dropdown-indicator"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
